pkg/virt-launcher/virtwrap/network/podinterface.go

```diff
@@ -735,3 +747,60 @@ func (s *SlirpPodInterface) setCachedInterface(name string) error {
 	err := writeToCachedFile(&s.domain.Spec.QEMUCmd.QEMUArg[s.podInterfaceNum], qemuArgCacheFile, name)
 	return err
 }
+
+type MacvtapPodInterface struct {
+	vmi              *v1.VirtualMachineInstance
+	vif              *VIF
+	iface            *v1.Interface
+	domain           *api.Domain
+	podInterfaceNum  int
+	podInterfaceName string
+}
+
+func (m *MacvtapPodInterface) discoverPodNetworkInterface() error {
+	if len(m.vif.MAC) == 0 {
+		// Get interface MAC address
+		mac, err := Handler.GetMacDetails(m.podInterfaceName)
+		if err != nil {
+			log.Log.Reason(err).Errorf("failed to get MAC for %s", m.podInterfaceName)
+			return err
+		}
+		m.vif.MAC = mac
+	}
+
+	return nil
+}
+
+func (m *MacvtapPodInterface) preparePodNetworkInterfaces() error {
+	return nil
```

Comment:
Don't we need to drop the original MAC from the pod interface after we assign it to the VM? Can it have the same MAC inside and outside?

Reply:
Yes, see https://bugzilla.redhat.com/show_bug.cgi?id=1723367#c13